release/rancher/rancher.go

```diff
@@ -296,3 +335,144 @@ func createPRFromRancher(ctx context.Context, rancherBaseBranch, title, branchNa
+	Content string
+}
+
+type Content struct {
+	RancherImages  []ContentLine
+	FilesWithRC    []ContentLine
+	MinFilesWithRC []ContentLine
+	ChartsWithDev  []ContentLine
+	KDMWithDev     []ContentLine
+}
+
+func CheckRancherRCDeps(forCi bool, org, repo, commitHash, files string) error {
+	var (
+		matchCommitMessage bool
+		badFiles           bool
+		content            Content
+	)
+
+	devDependencyPattern := regexp.MustCompile(`dev-v[0-9]+\.[0-9]+`)
+	rcTagPattern := regexp.MustCompile(`-rc[0-9]+`)
+
+	httpClient := ecmHTTP.NewClient(time.Second * 15)
+
+	if repo == "" {
+		repo = rancherRepo
+	}
+	if org == "" {
+		org = rancherOrg
+	}
+
+	//should return data if executed in rancher project root path
+	err := writeRancherImagesDeps(&content, rcTagPattern)
+	if err != nil {
+		return err
+	}
+
+	for _, filePath := range strings.Split(files, ",") {
+		repoContent, err := repository.ContentByFileNameAndCommit(org, repo, commitHash, filePath, &httpClient)
+		if err != nil {
+			return err
+		}
+
+		scanner := bufio.NewScanner(strings.NewReader(string(repoContent)))
+		lineNum := 1
+
+		for scanner.Scan() {
+			lineByte, line := formatLineByte(scanner.Text())
+
+			if devDependencyPattern.Match(lineByte) {
+				badFiles = true
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				lineContentLower := strings.ToLower(lineContent.Content)
+				if strings.Contains(lineContentLower, "chart") {
+					content.ChartsWithDev = append(content.ChartsWithDev, lineContent)
+				}
+				if strings.Contains(lineContentLower, "kdm") {
+					content.KDMWithDev = append(content.KDMWithDev, lineContent)
+				}
+			}
+			if strings.Contains(filePath, "/package/Dockerfile") {
+				if !strings.Contains(line, "_VERSION") {
+					continue
+				}
+				matches := regexp.MustCompile(`CATTLE_(\S+)_MIN_VERSION`).FindStringSubmatch(line)
+				if len(matches) == 2 && strings.Contains(line, "-rc") {
+					lineContent := ContentLine{Line: lineNum, File: filePath, Content: formatContentLine(line)}
+					content.MinFilesWithRC = append(content.MinFilesWithRC, lineContent)
+				}
+			}
+			if rcTagPattern.Match(lineByte) {
+				badFiles = true
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				content.FilesWithRC = append(content.FilesWithRC, lineContent)
+			}
+			lineNum++
+		}
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+	}
+
+	tmpl := template.New("rancher-release-rc-dev-deps")
+	tmpl = template.Must(tmpl.Parse(templateCheckRCDevDeps))
+	buff := bytes.NewBuffer(nil)
+	err = tmpl.ExecuteTemplate(buff, "componentsFile", content)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(buff.String())
+
+	if forCi && matchCommitMessage && badFiles {
+		return errors.New("check failed, some files don't match the expected dependencies for a final release candidate")
+	}
+
+	return nil
+}
+
+func writeRancherImagesDeps(content *Content, rcTagPattern *regexp.Regexp) error {
```

Comment:
This function could use documentation, and perhaps be renamed (it does not write to the files it opens).

Reply:
This func will be removed :D